### ISSUE TITLE
Question for implementing recovery

### ISSUE THREAD

**Issue by loloxwg** [cmu-db/bustub#778]
The Bustub instance does not seem to utilize log recovery. There is no initialization of a flush thread, and upon system startup, there are no actions related to undo or redo operations. 
Additionally, the current implementation of `table_page.cpp` does not include any checks for `if (enable_logging)`.
 It seems that in a certain commit, there was a comment like this:  
```  
/**
 * Removed to support new lock manager API for p4 (multilevel locking); Big hack energy
 * This clause was used in logging and recovery projects previously; not being used right now
 */
```
Why was it removed? It seems that `log_manager->AppendLogRecord` is not working now. 
For building a disk oriented db, recovery is important part, I think the recovery should be a part of project :)

**Comment by skyzh** [comment]
it's not supported anymore and the dead code should be cleaned up, maybe cc @yliang412?

**Comment by yliang412** [comment]
Yes, I agree recovery is a crucial part of any DBMS. The log recovery project, however, has not been a project since https://15445.courses.cs.cmu.edu/fall2019/project4/, and the codebase has had major changes over the semesters. The course staff currently do not plan to bring it back in Spring 2024 but will consider adding back support in a future semester.

**Comment by loloxwg** [comment]
Looking forward to seeing the recovery in the future. @yliang412
